### PR TITLE
Implement i2c operations for stm32f103xx

### PR DESCRIPTION
**Pull request by deadprogram**
This PR adds support for i2c to the stm32f103xx boards such as the "Bluepill".

In similar fashion to the UART and SPI interfaces, the stm32 starts numbering with `I2C1`, hence `I2C0` and `I2C1` both refer to the first interface. Support for the actual second interface `I2C2` will have to await a future PR.

### CODE REVIEW

**Comment by deadprogram** [comment]
Tested with Bluepill board and both the BlinkM code and this MPU6050 code:

```go
package main

import (
	"machine"
	"time"

	"github.com/aykevl/tinygo-drivers/mpu6050"
)

func main() {
	machine.I2C0.Configure(machine.I2CConfig{SDA: machine.PB9, SCL: machine.PB8, Frequency: 400000})

	// Init mpu6050
	accel := mpu6050.New(machine.I2C0)
	accel.Configure()

	for {
		x, y, z := accel.ReadAcceleration()
		println(x, y, z)
		time.Sleep(time.Millisecond * 100)
	}
}
```

File: src/machine/machine_stm32f103xx.go [hunk 1]
```diff
@@ -290,3 +291,463 @@ func (spi SPI) setPins(sck, mosi, miso uint8) {
...
+
+			// wait for stop
+			return i2c.waitForStop()
+		}
+	}
+
+	return nil
+}
+
+const TIMEOUT = 500
```
Comment (aykevl) [inline src/machine/machine_stm32f103xx.go]: This is in all-caps, meaning it is exported. Also, it is a global constant just for I2C without I2C in the name.
Can you rename it so it at least won't be exported?

File: src/machine/machine_stm32f103xx.go [hunk 2]
```diff
@@ -290,3 +291,463 @@ func (spi SPI) setPins(sck, mosi, miso uint8) {
...
+}
+
+// Send address of device we want to talk to
+func (i2c I2C) sendAddress(address uint8, write bool) error {
+	data := (address << 1)
+	if write {
+		data &^= stm32.I2C_OAR1_ADD0
+	} else {
+		data |= stm32.I2C_OAR1_ADD0
+	}
```
Comment (aykevl) [inline src/machine/machine_stm32f103xx.go]: I _guess_ this is here to set the read/write bit? If so, it is probably equivalent to the code [here](https://github.com/aykevl/tinygo/blob/master/src/machine/machine_atmega.go#L174).
Totally optional, but I think the code there is more readable.

File: targets/bluepill.json [hunk 3]
```diff
@@ -12,5 +12,7 @@
 	"extra-files": [
 		"src/device/stm32/stm32f103xx.s"
 	],
-	"flash": "openocd -f interface/stlink-v2.cfg -f target/stm32f1x.cfg -c 'program {hex} reset exit'"
+	"flash": "openocd -f interface/stlink-v2.cfg -f target/stm32f1x.cfg -c 'program {hex} reset exit'",
+	"ocd-daemon": ["openocd", "-f", "interface/stlink-v2.cfg", "-f", "target/stm32f1x.cfg"],
+	"gdb-initial-cmds": ["target remote :3333", "monitor halt", "load", "monitor reset", "c"]
```
Comment (aykevl) [inline targets/bluepill.json]: Ideally this change would go in with a separate commit, as it is not related to I2C support.

File: src/machine/machine_stm32f103xx.go (hunk 1 above)
Comment (deadprogram) [reply src/machine/machine_stm32f103xx.go]: Fixed.

File: src/machine/machine_stm32f103xx.go (hunk 2 above)
Comment (deadprogram) [reply src/machine/machine_stm32f103xx.go]: good call, I changed to match the more readable code.

File: targets/bluepill.json (hunk 3 above)
Comment (deadprogram) [reply targets/bluepill.json]: Agreed, I will make more effort to keep my PRs atomic.

**Comment by deadprogram** [comment]
All feedback has now been addressed.

**Comment by aykevl** [comment]
Merged in b8c326d710a88877413e550b06100d9b79373dcd.
